hist/histpainter/src/TGraphPainter.cxx

```diff
@@ -3522,96 +3646,72 @@ void TGraphPainter::PaintGraphErrors(TGraph *theGraph, Option_t *option)
       if (option3) {
          xline[if1-1] = x;
          xline[if2-1] = x;
-         yline[if1-1] = gPad->YtoPad(theY[i] + ey);
-         yline[if2-1] = gPad->YtoPad(theY[i] - ey);
+         yline[if1-1] = yup2;
+         yline[if2-1] = ylow2;
          if1++;
          if2--;
          continue;
       }
 
-      xl1 = x - s2x*cx;
-      xl2 = gPad->XtoPad(theX[i] - ex);
-      if (xl1 > xl2) {
+      if (ex) {
```

Comment:
@lmoneta: your previous comment apply also here. it should be `if (ex != 0.) {` . Thanks for having noticed it.

Reply:
done